tests/testthat/test-predict-boots.R

```diff
@@ -122,32 +129,71 @@ test_that("predict_boots() throws an error when bad n is specified", {
 })
 
 test_that("predict_boots() throws an error when training_data/new_data doesn't match expected format", {
+  skip_if_not_installed("xgboost")
+  skip_if_not_installed("parsnip")
+  suppressPackageStartupMessages(library(workflows))
+  suppressPackageStartupMessages(library(parsnip))
+
+  test_wf_fit <- fit(test_wf, test_train)
+
 
   # predictors & outcome missing from training_data
-  expect_error(
+  expect_snapshot(
     predict_boots(
-      workflow = test_wf,
+      workflow = test_wf_fit,
       n = 1,
       training_data = test_train[, 3],
       new_data = test_test
     ),
-
-    paste0("missing cols in training_data:\n",
-           "species, island, bill_length_mm, bill_depth_mm, flipper_length_mm, sex, body_mass_g")
+    error = TRUE
   )
 
   # predictors missing from new_data
-  expect_error(
+  expect_snapshot(
     predict_boots(
-      workflow = test_wf,
+      workflow = test_wf_fit,
       n = 1,
       training_data = test_train,
       new_data = test_test[, 3]
     ),
+    error = TRUE
+  )
+
+})
+
+test_that("predict_boots() can use formula interface", {
+  skip_if_not_installed("parsnip")
+  suppressPackageStartupMessages(library(workflows))
+  suppressPackageStartupMessages(library(parsnip))
+
+  car_subset <- mtcars[, c("mpg", "disp", "wt")]
+  lm_wflow <- workflow(mpg ~ ., parsnip::linear_reg())
+  lm_fit <- fit(lm_wflow, car_subset)
+
+  new_car <- data.frame(disp = 150.0, wt = 2.5)
 
-    paste0("missing cols in new_data:\n",
-           "species, island, bill_length_mm, bill_depth_mm, flipper_length_mm, sex")
+  # generate predictions
+  expect_warning(
+    x <-
+      predict_boots(
+        workflow = lm_fit,
+        n = 5,
+        training_data = car_subset,
+        new_data = new_car
+      ),
+
+    "At least 2000 resamples recommended for stable results."
   )
 
+  # tests
+  expect_s3_class(x, c("tbl_df", "tbl", "data.frame"))
```

Comment:
We could reduce a lot of these by making a "ptype" list which is just a zero-row slice of the data (e.g. `x[0,]`). That contains the classes of the overall object as well as the columns.